node/podcontroller.go

```diff
@@ -178,10 +178,18 @@ type PodControllerConfig struct {
 
 	// SyncPodsFromKubernetesRateLimiter defines the rate limit for the SyncPodsFromKubernetes queue
 	SyncPodsFromKubernetesRateLimiter workqueue.RateLimiter
+	// SyncPodsFromKubernetesShouldRetryFunc allows for a custom retry policy for the SyncPodsFromKubernetes queue
+	SyncPodsFromKubernetesShouldRetryFunc queue.ShouldRetryFunc
+
 	// DeletePodsFromKubernetesRateLimiter defines the rate limit for the DeletePodsFromKubernetesRateLimiter queue
 	DeletePodsFromKubernetesRateLimiter workqueue.RateLimiter
+	// DeletePodsFromKubernetesShouldRetryFunc allows for a custom retry policy for the SyncPodsFromKubernetes queue
+	DeletePodsFromKubernetesShouldRetryFunc queue.ShouldRetryFunc
```

Comment:
Can't use an internal type for external config.

Reply:
Can you describe the default behavior (when no retry func is provided) in the doc string?

Reply:
Where should I declare it?

Reply:
The only thing I can think of (that wouldn't introduce an import loop), is declare a new API package called "queue" under root and declare the ShouldRetryFunc there. Opinion?

Reply:
Or define 2 types one for external one for internal.

Reply:
Did you have a concern with defining the type twice? I see there's a new package now.

Reply:
I do not think we should define types twice. It goes against DRY.

Reply:
Seems like the code contortions is worse than a repeat :(
And if there's ever a change it will fail to compile anyway.

Reply:
What do you think of moving the queue package out of internal?

Reply:
Also, wait, are you sure if I "re-export" the tag in podcontroller.go, it doesn't work? A la:

var ShouldRetryFunc = queue.ShouldRetryFunc

...

Reply:
That may work well (or rather `type ShouldRetryFunc = queue.ShouldRetryFunc`

Reply:
Done.

Reply:
Checked this on godoc, looks good.